src/EditorFeatures/Core/Diagnostics/AbstractDiagnosticsTaggerProvider.cs

```diff
@@ -110,4 +77,33 @@ private static ITaggerEventSource CreateEventSourceWorker(ITextBuffer subjectBuf
     /// <returns>an array of locations that should have the tag applied.</returns>
     protected virtual ImmutableArray<DiagnosticDataLocation> GetLocationsToTag(DiagnosticData diagnosticData)
         => diagnosticData.DataLocation is not null ? ImmutableArray.Create(diagnosticData.DataLocation) : ImmutableArray<DiagnosticDataLocation>.Empty;
+
+    public ITagger<T>? CreateTagger<T>(ITextBuffer buffer) where T : ITag
```

Comment:
this code was pulled up from AbstractPushOrPullDiagnosticsTaggerProvider.PullDiagnosticsTaggerProvider.cs (which was deleted and merged into the parent type).